cats/cats-redis/src/test/java/com/netflix/spinnaker/cats/redis/cluster/ClusteredSortAgentSchedulerTest.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.cats.redis.cluster;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.netflix.spinnaker.cats.agent.CachingAgent;
+import com.netflix.spinnaker.cats.agent.DefaultCacheResult;
+import com.netflix.spinnaker.cats.agent.ExecutionInstrumentation;
+import com.netflix.spinnaker.cats.cluster.DefaultAgentIntervalProvider;
+import com.netflix.spinnaker.cats.test.TestAgent;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.commons.lang3.reflect.MethodUtils;
+import org.junit.Before;
+import org.junit.Test;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+
+public class ClusteredSortAgentSchedulerTest {
+
+  private ClusteredSortAgentScheduler clusteredSortAgentScheduler;
+
+  private Jedis jedis = mock(Jedis.class);
+  private JedisPool jedisPool = mock(JedisPool.class);
+  private Integer parallelism = 2;
+  private CachingAgent.CacheExecution agentExecution = mock(CachingAgent.CacheExecution.class);
+  private ExecutionInstrumentation executionInstrumentation = mock(ExecutionInstrumentation.class);
+
+  private String WAITING_SET;
+  private String WORKING_SET;
+  private Method saturatePoolMethod;
+  private Optional<Semaphore> runningAgents;
+
+  @Before
+  public void setUp() throws IllegalAccessException {
+    when(jedisPool.getResource()).thenReturn(jedis);
+    when(jedis.scriptLoad(anyString())).thenReturn("testScriptSha");
+    when(jedis.time()).thenReturn(List.of("1678784468", "374338"));
+    DefaultAgentIntervalProvider intervalProvider = new DefaultAgentIntervalProvider(6000000);
+    clusteredSortAgentScheduler =
+        new ClusteredSortAgentScheduler(jedisPool, () -> false, intervalProvider, parallelism);
+
+    WAITING_SET =
```

Comment:
Easier to make these fields package private in ClusteredSortAgentScheduler?  Marking them as `@VisibleForTesting` makes it pretty clear why...then we don't need a copy here.

Reply:
Ok, I get it, thanks!